Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-programming/questions-programming.component.ts

```diff
@@ -27,35 +30,86 @@ export class QuestionsProgrammingComponent implements OnInit {
     isCategoryReady: boolean;
     isLanguageReady: boolean;
     isFormSubmitted: boolean;
+    isQuestionEdited: boolean;
     code: any;
     testCases: CodeSnippetQuestionsTestCases[];
     //To enable enum testCaseType in template
     testCaseType: TestCaseType;
-    max: number;
+    questionId: number;
 
     private successMessage: string = 'Question saved successfully';
     private failedMessage: string = 'Question failed to save';
-    private routeToDashboard = '/questions';
+    private routeToDashboard = ['questions'];
 
     constructor(private questionsService: QuestionsService,
         private categoryService: CategoryService,
         private snackBar: MdSnackBar,
-        private router: Router) {
+        private router: Router,
+        private route: ActivatedRoute) {
 
         this.nolanguageSelected = true;
         this.isCategoryReady = false;
         this.isLanguageReady = false;
+        this.isQuestionEdited = false;
         this.selectedLanguageList = new Array<string>();
         this.codingLanguageList = new Array<string>();
         this.categoryList = new Array<Category>();
         this.questionModel = new QuestionBase();
+        this.selectedCategory = 'Please select a category';
+        this.selectedDifficulty = 'Easy';
         this.formControlModel = new FormControlModel();
         this.testCases = new Array<CodeSnippetQuestionsTestCases>();
     }
 
     ngOnInit() {
-        this.getCodingLanguage();
-        this.getCategory();
+        this.questionId = this.route.snapshot.params['id'];
+
+        if (!this.questionId) {
+            this.getCodingLanguage();
+            this.getCategory();
+        }
+        else {
+            this.isQuestionEdited = true;
+            this.prepareToEdit(+this.questionId);
```

Comment:
The null case for `questionId` is being handled in if condition at line 67 - `if (!this.questionId)`. If `questionId` is null it will enter into the `if` block and the page will be rendered for adding new code snippet question instead of editing code snippet question.